src/libraries/Microsoft.PowerFx.Core/Types/DType.cs

```diff
@@ -3535,5 +3554,28 @@ public bool TryGetSimilarName(DName name, FieldNameKind fieldNameKind, out strin
             return similar != null &&
                    comparer.Distance(similar) < (name.Value.Length / 3) + 3;
         }
+
+        // Determins if the result of a numeric binary operation should be a Decimal or a Number.
+        // If !numberIsFloat then Strings and Booleans are allowed to coerce to Decimal for this test, as if they were passed through Value
+        public static bool DecimalBinaryOp(DType leftType, DType rightType, bool numberIsFloat)

```

Comment:
probably move to BindingHelper/BinderUtils,

Also, could be named something like: `IsBinaryOpResultDecimal`